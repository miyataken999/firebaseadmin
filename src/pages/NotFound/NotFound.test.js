import React from 'react';

import { fireEvent } from '@testing-library/react';
import NotFound from '.';
import paths from '../Router/paths';

describe('<NotFound /> rendering', () => {
  const location = {
    pathname: '/',
  };

  it('should render without crashing', () => {
    const { component } = renderWithProviders(<NotFound location={location} />)(
      {
        auth: {
          userData: {},
        },
      }
    );

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should display the button with the login path if user is not authenticated', () => {
    const { component } = renderWithProviders(<NotFound location={location} />)(
      {
        auth: {
          userData: {
            id: null,
          },
        },
      }
    );
    fireEvent.click(component.getByRole('link'));
    expect(window.location.pathname).toBe(paths.LOGIN);
  });

  it('should display the button with the home path if user is authenticated', () => {
    const { component } = renderWithProviders(<NotFound location={location} />)(
      {
        auth: {
          userData: {
            id: 'some userId',
          },
        },
      }
    );

    fireEvent.click(component.getByRole('link'));
    expect(window.location.pathname).toBe(paths.ROOT);
  });
});
