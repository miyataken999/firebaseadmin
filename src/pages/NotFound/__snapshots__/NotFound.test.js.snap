// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> rendering should render without crashing 1`] = `
<DocumentFragment>
  <section
    class="hero is-fullheight"
  >
    <div
      class="hero-body"
    >
      <section
        class="section section"
      >
        <div
          class="container"
        >
          <div
            class="columns is-vcentered is-desktop"
          >
            <div
              class="column has-text-centered"
            >
              <h1
                class="title"
              >
                Error 404: page not found
              </h1>
              <p
                class="subtitle"
              >
                The requested URL / was not found
              </p>
              <a
                class="button is-info is-normal"
                href="/login"
              >
                Go Back
              </a>
            </div>
            <div
              class="column has-text-centered"
            >
              <img
                alt="404 error"
                src="404.gif"
              />
            </div>
          </div>
        </div>
      </section>
    </div>
  </section>
</DocumentFragment>
`;
